src/win32.js

```diff
@@ -98,15 +99,53 @@ class WindowsApp extends App {
     }
   }
 
+  async signAppIfSpecified () {
+    const windowsSignOpt = this.opts.windowsSign
+    const windowsMetaData = this.opts.win32metadata
+
+    if (windowsSignOpt) {
+      const signOpts = createSignOpts(windowsSignOpt, windowsMetaData, this.renamedAppPath)
+      debug(`Running @electron/windows-sign with the options ${JSON.stringify(signOpts)}`)
+      try {
+        await sign(signOpts)
+      } catch (err) {
+        // Although not signed successfully, the application is packed.
+        if (signOpts.continueOnError) {
+          common.warning(`Code sign failed; please retry manually. ${err}`, this.opts.quiet)
+        } else {
+          throw err
+        }
+      }
+    }
+  }
+
   async create () {
     await this.initialize()
     await this.renameElectron()
     await this.copyExtraResources()
     await this.runRcedit()
+    await this.signAppIfSpecified()
     return this.move()
   }
 }
 
+function createSignOpts (properties, windowsMetaData, appDirectory) {
+  let result = { appDirectory }
+
+  if (typeof properties === 'object') {
+    result = { ...properties, appDirectory }
+  }
+
+  // A little bit of convenience
+  if (windowsMetaData) {
+    if (windowsMetaData.FileDescription) {
+      result.description = windowsMetaData.FileDescription
+    }
+  }
```

Comment:
Should this really supercede the user provided value? Probably should add a `!description` check to avoid overwriting.  Also collapse these if statements

```suggestion
  // A little bit of convenience
  if (windowsMetaData && windowsMetaData.FileDescription && !result.description) {
    result.description = windowsMetaData.FileDescription
  }
```